### PR TITLE
[202205] Disable interrupt for intel i2c-i801 driver

### PR DESCRIPTION
**Pull request by prgeor**
#### Why I did it

On S6100 we are seeing almost 100K interrupts per second on intels i801 SMBUS controller which affects systems performance. This interrupt storm has already been fixed in newer kernel with a fix here:-

In 202205 branch (using old kernel (5.10)) disable the interrupt and enable the polling for i2c-i801 driver

image

##### Work item tracking
- Microsoft ADO **(number only)**:24910530

#### How I did it
Disable the interrupt by passing the interrupt disable feature argument to i2c-i801 driver

#### How to verify it
1. On SN2700 its already disabled in Mellanox hw-mgmt
2. Arista 7050CX3 don't use i801 
3. Celestica DX010 
4. Dell S6100 verified the interrupts are no longer incrementing.
5. 

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ] 

#### Description for the changelog

Disable interrupt for intel i2c-i801 driver

### CODE REVIEW

**Comment by prgeor** [comment]
@Staphylo can you review from Arista platform pov?

**Comment by yxieca** [comment]
@prgeor should there be a master branch PR for this change and we cherry-pick it?

**Comment by prgeor** [comment]
> @prgeor should there be a master branch PR for this change and we cherry-pick it?

@yxieca master already has latest kernel where this interrupt is by default disabled

**Review by Staphylo — CHANGES_REQUESTED** [review]
Hi @prgeor, your change should be fine for our platforms.
However I noticed a syntax error that will break `logs_inram`

On a more general note with this change:
I haven't seen any high interrupt issue on this product. Also I'm not sure where the statement that disabling the IRQ became the default. Looking at the kernel sources the FEATURE_IRQ is enabled on the latest 5.10 and mainline kernel.
The performance is slightly degraded after disabling this feature.
However there should not be any impact to our software since we don't use this smbus controller for much.

File: files/Aboot/boot0.j2 [hunk 1]
```diff
@@ -503,7 +503,7 @@ write_platform_specific_cmdline() {
     if [ "$sid" = "Gardena" ] || [ "$sid" = "GardenaE" ]; then
         aboot_machine=arista_7260cx3_64
         flash_size=28000
-        cmdline_add logs_inram=on
+        cmdline_add logs_inram=on, i2c-i801.disable_features=0x10
```
[the remark this answers is not in the document]
Comment (prgeor) [reply files/Aboot/boot0.j2]: @Staphylo added in new line

**Comment by prgeor** [comment]
@yxieca Just realized, this fix needs to be in master

> > @prgeor should there be a master branch PR for this change and we cherry-pick it?
> 
> @yxieca master already has latest kernel where this interrupt is by default disabled

@yxieca on a second look, looks like we need PR in master. So closing this

**Pull request closed without merging**